### DISCUSSION TITLE
Wheel alignment question

### DISCUSSION THREAD

**Discussion by tfield** [BroncBotz3481/YAGSL-Example#119] [Q&A] [answered]
Hi team - question… we are reworking our conversion factor and have already double checked our wheel locations already… but we are seeing this behaviour from yagsl for neo/mk4i. (We started with a good known config).  What is causing the front wheels to angle in and the back wheels to angle out for all directions? We’ll run with any pointers you can give! :-)

https://youtube.com/shorts/NZS0mBtpnOs?si=WT6ggLaooA6kpsiT

**Comment by thenetworkgrinch** [comment, marked as answer]
Great video! 

Looking at it I can clearly see your conversion factor is off or possibly your cancoders are not configured for the right modules. YAGSL is commanding it to go the correct directions (as noted by the red lines). 

For the conversion factor i just made some more documentation for that here https://yagsl.gitbook.io/yagsl/fundamentals/swerve-modules#conversion-factor 

Just so you are aware, use `1` as the `encoderPulsePerRevolution` im `SwerveMath` functions calculating the conversion factors. TalonFX's were updated in 2024 for this to be possible, however if you are using a TalonFX steer/azimuth/angle motor wuth the `2023` YAGSL you need to use `2048` for the `encoderPulsePerRevolutoon`

**Reply by tfield** [reply]
That was it!! Thank you!! 

We will keep tuning it but we will be happy to share our known config if that is helpful. 

https://youtube.com/shorts/XXej0P76LUY?si=ZN4tlAjWVJ7IFlSX

**Reply by thenetworkgrinch** [reply]
Absolutely!! Can you make a PR for it on the repo?

**Reply by aliddy23** [reply]
For 2023 with Falcon steer motors, would it not be `2048` for `encoderPulsePerRevolution`? @thenetworkgrinch

**Reply by thenetworkgrinch** [reply]
It would be, thank you for catching that! @aliddy23
